src/aind_metadata_service/sharepoint/las2020/mapping.py

```diff
@@ -1873,10 +1915,25 @@ def aind_whereto_obtainsubstance_i(self) -> Optional[Any]:
             }.get(self._las.whereto_obtainsubstance_i, None)
         )
 
+    def has_ip_injection(self) -> bool:
+        """Is there a headpost procedure?"""
+        return (
+            LASProcedure.DOSING
+            in [self.aind_req_pro1, self.aind_req_pro2, self.aind_req_pro3]
+            and self.aind_dose_route == Doseroute.INTRAPERITONEAL_IP
+        )
+
     def get_procedure(self) -> Surgery:
         """Return Surgery as best as possible from a record."""
-        # TODO: add RO Injection and IP Injections to procedures
+        # TODO: add RO Injection to procedures
         subject_procedures = []
+        if self.has_ip_injection():
+            # TODO: map injection_materials, protocol_id
```

Comment:
is there a separate ticket for this? IMO this is a key part of this ticket so I'd prefer it be done here, but if there's a good reason to do it separately we need to make sure it gets done.

Reply:
Protocol_ids are integrated via the smartsheet endpoint, but there is no "IP injection" or "LAS Surgery" in the smartsheet Published Protocols list. We'll need to look at examples of the "substance to be administered" field to determine how to map injection_materials